pkg/tar/tar_test.go

```diff
@@ -185,23 +187,60 @@ func TestCreateTar(t *testing.T) {
 	verifyTarFile(t, tarFile, testFiles, testLinks)
 }
 
-func createTestTar(files []fileDesc, writer io.Writer) {
+func createTestTar(files []fileDesc, writer io.Writer, t *testing.T) {
 	tw := tar.NewWriter(writer)
+	defer tw.Close()
 	for _, fd := range files {
-		contentBytes := []byte(fd.content)
-		hdr := &tar.Header{
-			Name:       fd.name,
-			Mode:       int64(fd.mode),
-			Size:       int64(len(contentBytes)),
-			Typeflag:   tar.TypeReg,
-			AccessTime: time.Now(),
-			ModTime:    fd.modifiedDate,
-			ChangeTime: fd.modifiedDate,
+		if isSymLink(fd.mode) {
+			if err := addSymLink(tw, &fd, t); err != nil {
+				t.Fatalf(
+					"Unable to add symbolic link %s (that points to %s) to archive: %v",
+					fd.name, fd.target, err,
+				)
+			}
+		} else {
+			addRegularFile(tw, &fd)
 		}
-		tw.WriteHeader(hdr)
-		tw.Write(contentBytes)
 	}
-	tw.Close()
+}
+
+func addRegularFile(tw *tar.Writer, fd *fileDesc) {
+	contentBytes := []byte(fd.content)
+	hdr := &tar.Header{
+		Name:       fd.name,
+		Mode:       int64(fd.mode),
+		Size:       int64(len(contentBytes)),
+		Typeflag:   tar.TypeReg,
+		AccessTime: time.Now(),
+		ModTime:    fd.modifiedDate,
+		ChangeTime: fd.modifiedDate,
+	}
+	tw.WriteHeader(hdr)
```

Comment:
It's because I tried to don't touch original code (I only extracted it to the separate method). But, I agree with you and I will add check here too. Thank you!